docs/src/modules/spring/pages/actions-publishing-subscribing.adoc

```diff
@@ -1 +1,100 @@
+
+
+== Subscribing to a Topic
+
+To receive messages from a from Google Cloud Pub/Sub or Apache Kafka topic, annotate the service method `@Subscribe.Topic` and specify the topic name.
+
+The events from the topic are delivered to the Action and in this simplified example they are only logged.
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=sub-topic-action]
+----
+<1> subscribing to the events from topic 'counter-events'. 
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE in Subscribing to an Event Sourced Entity.
+
+
+Note that by default, Kalix assumes the message in the topic was serialized as JSON and as such deserialize it into a `CounterEvent`. You can find more about serialization in Kalix here xref:spring:serialization.adoc[]
+
+== Receiving messages from an external Topic
+
+In the above example, you consumed JSON messages from a topic that you control. If you are consuming an external topic, the message format may not be under your control and may not be JSON.
+
+In such case, the Action definition should receive an array of bytes and deserialization must be handled within the method. See xref:spring:serialization.adoc[Handling Serialization] for more information on how to deal with data formats.
+
+== Subscribing and acting upon
+
+Another possible usage for Actions is to consume events and act upon.
+
+For example, you may consume events from one entity or from a topic, transform to commands and send to an another entity or an external system. This is similar to the usage explained in xref:actions-as-controller.adoc[Actions as Controller], except that the Action is driven by the flow of incoming events instead of external user requests.
+
+== Publishing events to a Topic
+
+To publish events in a topic you need also subscribe to an entity or a topic. Let's assume that you have the counter entity mentioned above. Then you can subscribe to the events of this counter, transform them and then publish them in a topic. 
+
+To publish the events you need to add the annotation `@Publish.Topic` to the method subscribed to the events and add the name of the topic. 
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=sub-ESE-pub-topic-action]
+----
+<1> subscribing to the events from the Counter.
+<2> publishing to a topic name 'counter-events'.
+<3> setting the method input type to the events produced by the counter.
+<4> any return is valid. See NOTE in Subscribing to an Event Sourced Entity.
+
+Note that the messages stored in the topic are serialized as JSON. You cannot change this serialization.
```

Comment:
Maybe use `IMPORTANT:` here?

```suggestion
IMPORTANT: the messages stored in the topic are serialized as JSON. You cannot change this serialization.
```

Reply:
Yeah, this is a good point my only concern when writing this and the 'Note that' above - that you suggested adding a NOTE: - is that it will lead the code with no leading text and looks somehow abrupt. But I agree it's important the reader notes it.